atomic_reactor/plugins/post_rpmqa.py

```diff
@@ -70,3 +87,49 @@ def gather_output(self):
                 return output
 
         raise RuntimeError('Unable to gather list of installed packages in container')
+
+    def gather_output_scratch(self):
+        rpmdb_dir = tempfile.mkdtemp()
+        container_dict = self.tasker.create_container(self.image_id, command=['/bin/bash'])
+        container_id = container_dict['Id']
+        self._container_ids.append(container_id)
+
+        try:
+            bits, _ = self.tasker.get_archive(container_id, RPMDB_PATH)
+        except APIError as ex:
+            self.log.info('Could not extract rpmdb in %s : %s', RPMDB_PATH, ex)
+            return None
+        except Exception as ex:
+            self.log.info('Get archive failed while extracting rpmdb in %s : %s', RPMDB_PATH, ex)
+            raise RuntimeError('%s' % ex)
+
+        with tempfile.NamedTemporaryFile() as rpmdb_archive:
+            for chunk in bits:
+                rpmdb_archive.write(chunk)
+            rpmdb_archive.flush()
+            tar_archive = tarfile.TarFile(rpmdb_archive.name)
+
+        tar_archive.extractall(rpmdb_dir)
+
+        rpmdb_path = os.path.join(rpmdb_dir, RPMDB_DIR_NAME)
+        rpmdb_packages = os.path.join(rpmdb_path, RPMDB_PACKAGES_NAME)
+
+        if not os.path.exists(rpmdb_packages):
+            self.log.info('%s does not exist in rpmdb', RPMDB_PACKAGES_NAME)
+            return None
+
+        rpm_cmd = ['rpm', '--dbpath', rpmdb_path]
+        rpm_cmd.extend(rpm_qf_args(output_list=True))
+
+        try:
+            self.log.info('getting rpms from rpmdb: %s', rpm_cmd)
+            rpm_output = subprocess.check_output(rpm_cmd)
+        except Exception as e:
+            self.log.error("Failed to get rpms from rpmdb: %s", e)
+            raise e
+
+        if isinstance(rpm_output, bytes):
+            rpm_output = rpm_output.decode("utf-8")  # py2 & 3 compat
```

Comment:
We don't need to do this by passing `universal_newlines`  to `check_output`.

Reply:
ok